tests/test_optimizely_config.py

```diff
@@ -863,3 +902,137 @@ def test__get_events(self):
+                                        "name": "test_attribute",
+                                        "type": "custom_attribute",
+                                        "value": "test_value_1"
+                                    }
+                                 ]
+                             ]
+                         ]
+                ]
+            }
+        ]
+
+        self.assertEqual(expected_value, config.get_audiences())
+
+    def test_stringify_conditions(self):
+        '''
+            Test to confirm converting from audienceConditions list to String represented conditions
+            Note* this also test lookup_name_from_id in OptimizelyConfig
+        '''
+
+        experiment = {'audienceConditions': ['and', ['or', '3468206642', '3988293898'], [
+            'or', '3988293899', '3468206646', '3468206647', '3468206644', '3468206643']],
+            'audienceIds': ['0'], 'forcedVariations': {}, 'id': '1323241598'}
+
+        audience_conditions = experiment.get('audienceConditions')
+
+        audiences_map = {
+            '3468206642': 'us',
+            '3988293898': 'female',
+            '3988293899': 'male',
+            '3468206646': 'them',
+            '3468206647': 'anyone'
+        }
+
+        config = optimizely_config.OptimizelyConfig(
+            revision='101',
+            experiments_map={},
+            features_map={},
+            environment_key='TestEnvironmentKey',
+            attributes={},
+            events={},
+            audiences=None
+        )
+
+        config_service = optimizely_config.OptimizelyConfigService(config)
+
+        result = config_service.stringify_conditions(audience_conditions, audiences_map)
+
+        expected_result = '(("us" OR "female") AND ("male" OR "them" OR "anyone" OR "3468206644" OR "3468206643"))'
+
+        self.assertEqual(result, expected_result)
+
+    def test_update_experiment(self):
+        ''' Test that OptimizelyExperiment updates with proper conditions for audiences '''
+        ent_experiment = entities.Experiment(
+            '12345',
+            'test_update_experiment',
+            'Running',
+            variations=None,
+            forcedVariations=None,
+            trafficAllocation=None,
+            layerId=None,
+            audienceIds=['432', '321', '543'],
+            audienceConditions=['and', ['or', '432', '321'], '543']
+        )
+
+        audiences_map = {
+            '432': 'us',
+            '321': 'female',
+            '543': 'adult'
+        }
+
+        update_experiment = optimizely_config.OptimizelyExperiment(
+            '12345',
+            'test_update_experiment',
+            variations_map={}
+        )
+
+        config = optimizely_config.OptimizelyConfig(
+            revision='101',
+            experiments_map={},
+            features_map={},
+            environment_key='TestEnvironmentKey',
+            attributes={},
+            events={},
+            audiences=None
+        )
```

Comment:
Expect to make "audiences" and pass it as creating OptimizelyExperiment.

Reply:
Does this mean we want to create an experiments map and pass the audiences that way?